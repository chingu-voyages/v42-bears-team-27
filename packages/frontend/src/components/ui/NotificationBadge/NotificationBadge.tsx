export interface NotificationBadgeProps {
  /**
   * val to be displayed in badge
   */
  val?: number;
}

export const NotificationBadge: React.FC<NotificationBadgeProps> = ({
  val,
}) => (
  <div
    sx={{
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      width: 'max-content',
      minWidth: '1.5em',
      height: '1.5em',
      bg: '#dc4531',
      color: '#ffffff',
      border: 'none',
      borderRadius: '50%',
      cursor: 'pointer',
      fontWeight: '500',
    }}
  >
    <span>{val}</span>
  </div>
);

if (process.env.NODE_ENV !== 'production') {
  NotificationBadge.displayName = 'NotificationBadge';
}
