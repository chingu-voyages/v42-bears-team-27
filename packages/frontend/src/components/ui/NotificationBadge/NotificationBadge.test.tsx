import { render } from 'test-utils';

import { NotificationBadge } from './NotificationBadge';

describe('<NotificationBadge />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<NotificationBadge />);
      expect(container).toMatchSnapshot();
    });
  });
});
