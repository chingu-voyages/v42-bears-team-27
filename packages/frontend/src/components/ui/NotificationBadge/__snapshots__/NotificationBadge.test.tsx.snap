// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationBadge /> Snapshots should match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-width: 1.5em;
  height: 1.5em;
  background-color: #b31e1e;
  color: #ffffff;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  font-weight: 500;
}

<div>
  <div
    class="emotion-0"
  >
    <span />
  </div>
</div>
`;
